Documentation for updates to Umbraco Commerce shipping (#5802)
umbraco/UmbracoDocs: 13/umbraco-commerce/release-notes/v13.1.0-rc.md

### Shipping Packing Factories

Another important aspect of shipping calculations is defining packages along with their dimensions and weight. Umbraco Commerce comes with an out of the box package algorythm, but with products being stackable in many different ways, we've made this pluggable too to allow you to create more specific packaging logic.

See the [Shipping Packing Factories](../key-concepts/shipping-package-factories.md) concept documentation for more details.

### Measurements Property Editor / Store Measurement System

With the shipping packing factories it also becomes necesarry to capture the phsyical dimensions and weight of the individual products. To help with this, Umbraco Commerce now comes with a Measurements property editor to allow you to define all the physical attributes of your products.

![Measurements](../media/measurements.png)

We've also made it so that you can configure you prefered measurement system (metric or imperial) via a setting on the Store node, with the new Measurements property editor taking this into account.

### C# / Storefront API Updates

> A fundamental change that arrises with the new dynamic and realtime shipping rates is that it is now no longer possible to just retrieve a shipping rate from a shipping method, without a reference order. Addditionally, many shipping operators offer different products and services which means that shipping methods are now able to return multiple rate values for a given rate request.

Reviewer: ```suggestion
There is a fundamental change that arises with the new dynamic and real-time shipping rates. And that is that it's now no longer possible to retrieve a shipping rate from a shipping method, without a reference order. Additionally, many shipping operators offer different products and services. This means that shipping methods are now able to return multiple rate values for a given rate request.
```

> To accomodate this some API's have been, such as the shipping method `CalculatePrices` method is being replaced with `TryCalculateRates` that accepts an order, and can return multiple rates. In addition, we've also updated the `SetShippingMethod` API on the order to support passing in a `ShippingOption` to identify which rate option from the shipping operator is being selected.
> 
> These changes have also been made to the Storefront API to with some older endpoints becoming depricated, and new ones being created for these new approaches.
> 
> To maintain backwards compatibility, the old API's are still present, but these will only work with / return fixed rate shipping method definitions.
> 
> ## Add-on Updates
> 
> In addition to the core Umbraco Commerce changes, updates to the following add-on packages will also be made:
> 
> * **Umbraco.Commerce.Deploy** - Supports the store and shipping method udpates as well as the new locations model.
> * **Umbraco.Commerce.Checkout** - Add's support for displaying / selecting shipping rates on the front end. 

Reviewer: ```suggestion
To accommodate this some APIs have been, such as the shipping method `CalculatePrices` method is being replaced with `TryCalculateRates` that accepts an order, and can return multiple rates. In addition, we've also updated the `SetShippingMethod` API on the order to support passing in a `ShippingOption` to identify which rate option from the shipping operator is being selected.

These changes have also been made to the Storefront API with some older endpoints becoming deprecated, and new ones being created for these new approaches.

To maintain backward compatibility, the old APIs are still present, but these will only work with return fixed-rate shipping method definitions.

## Add-on Updates

In addition to the core Umbraco Commerce changes, updates to the following add-on packages will also be made:

* **Umbraco.Commerce.Deploy** - Supports the store and shipping method updates as well as the new locations model.
* **Umbraco.Commerce.Checkout** - Adds support for displaying/selecting shipping rates on the front end. 
```
